test/test_integration.py

```diff
@@ -958,16 +1004,15 @@ def socket_handler(listener):
 
             receive_preamble(sock)
 
+            # Wait for the message from the main thread.
+            send_event.wait()
             # Send the headers for the response. This response has no body.
             f = build_headers_frame(
                 [(':status', '200'), ('content-length', '0')]
             )
             f.flags.add('END_STREAM')
             f.stream_id = 1
             sock.sendall(f.serialize())
-
-            # Wait for the message from the main thread.
-            send_event.wait()
```

Comment:
We have to wait here only when the client side are going to get the response. Another option is add `send_event.wait()` here, and add `get_response()`, `send_event.set()` in client side.